client/signup/controller.js

```diff
@@ -63,10 +63,13 @@ const removeWhiteBackground = function () {
 	document.body.classList.remove( 'is-white-signup' );
 };
 
-const gutenbergRedirect = function () {
+const gutenbergRedirect = function ( flowName ) {
 	const url = new URL( window.location );
-	url.pathname = '/new';
-
+	if ( [ 'beginner', 'personal', 'premium', 'business', 'ecommerce' ].includes( flowName ) ) {
+		url.pathname = `/new/${ flowName }`;
```

Comment:
I think this might open a door for XSS. `flowName` is an unescaped query param that we redirect to. `encodeURIComponent` would probably do.

Reply:
The `includes` check takes care of it already, but good note otherwise. :-)

Reply:
Yup 🤦